docs/maintaining/testing.rst

```diff
@@ -0,0 +1,57 @@
+Testing
```

Comment:
Opsdroid *currently* uses markdown for all documentation. I'm happy to open up discussions around changing that, but I'd rather not introduce inconsistencies here without a change of policy.

Reply:
How's a policy on "use markdown, unless a sphinx extension breaks it".

Reply:
In this situation it sounds like the problem is with the extension. If an extension has a bug which results in not working correctly with `eval_rst` then I suggest we do not use the extension.

Reply:
The bug has been patched in https://github.com/astropy/sphinx-automodapi/pull/112. I have removed automodapi from this for now.